src/assisted_installer_controller/operator_handler.go

```diff
@@ -145,14 +155,97 @@ func (handler ClusterServiceVersionHandler) IsInitialized() bool {
 	if err != nil {
 		return false
 	}
+	handler.log.Infof("Operator %s csv name is %s", handler.GetName(), csvName)
+	if _, err := handler.kc.GetCSV(handler.operator.Namespace, csvName); err != nil {
+		if apierrors.IsNotFound(err) {
+			handler.log.WithError(err).Warnf("CSV %s not exists for %s. "+
+				"Going to handle it with OLMEarlySetupBug fix", csvName, handler.GetName())
+
+			err = handler.handleOLMEarlySetupBug()
+			if err != nil {
+				handler.log.WithError(err).Warnf("Failed to handle OLMEarlySetupBug")
+			}
+		} else {
+			handler.log.WithError(err).Warnf("Failed to get csv %s not for %s. ", csvName, handler.GetName())
+		}
 
-	if csvName == "" {
 		return false
 	}
 
 	return true
 }
 
+// handleOLMEarlySetupBug cleans failed olm jobs as they will rerun by olm and
+// deletes install plans to force olm re-conciliation and job recreation.
+// Assisted Controller to take care of it once the cluster is up. The service
+// does this because subscriptions created too early in the installation
```

Comment:
This comment is outdated as well

Reply:
why? i updated it.

Reply:
Because it talks about the service doing things which it no longer does

Reply:
fixed